merlin/systems/workflow/base.py

```diff
@@ -105,7 +109,16 @@ def run_workflow(self, input_tensors):
         if kind != Supports.CPU_DICT_ARRAY:
             transformed, kind = convert_format(transformed, kind, Supports.CPU_DICT_ARRAY)
 
-        return TensorTable(transformed).to_dict()
+        output_table = TensorTable(transformed)
+
+        for col in self.workflow.output_schema:
+            if col.is_ragged and output_table[col.name].offsets is None:
+                values, offsets = _to_ragged(output_table[col.name].values)
+                values = values.astype(self.output_dtypes[col.name + "__values"])
```

Comment:
adding back the dtype coercion that was removed in #317 to handle the case where some workflows output a dtype that is inconsistent with the `workflow.output_schema`.